apis/chef/src/test/java/org/jclouds/chef/domain/RoleTest.java

```diff
@@ -0,0 +1,117 @@
+      Role.builder().envRunList("does.not.matter", (List<String>) null);
+   }
+
+   @Test(expectedExceptions = NullPointerException.class)
+   public void canNotUseNullEnvNameWhenAddingEnvRunListEntry() {
+      Role.builder().envRunListElement((String) null, "does.not.matter");
+   }
+
+   @Test(expectedExceptions = NullPointerException.class)
+   public void canNotUseNullEntryWhenAddingEnvRunListEntry() {
+      Role.builder().envRunListElement("does.not.matter", (String) null);
+   }
+
+   public void multipleEnvRunListsCanBePopulated() {
+      String env1 = "env1";
+      String env2 = "env2";
+      String env1Alpha = "env1.alpha";
+      String env2Alpha = "env2.alpha";
+      String env2Bravo = "env2.bravo";
+      Role role = Role.builder().envRunListElement(env1, env1Alpha).envRunListElement(env2, env2Alpha)
+            .envRunListElement(env2, env2Bravo).build();
+      Map<String, List<String>> envRunList = role.getEnvRunList();
+      assertNotNull(envRunList, "envRunList");
+      assertEquals(envRunList.size(), 2, "envRunList.size");
+
+      verifyRunListForEnvironment(envRunList, env1, env1Alpha);
+      verifyRunListForEnvironment(envRunList, env2, env2Alpha, env2Bravo);
+   }
+
+   @Test(expectedExceptions = UnsupportedOperationException.class)
+   public void envRunListOnNewlyBuiltRoleIsImmutable() {
+      String env = "env";
+      Role role = Role.builder().envRunListElement(env, env + "1").build();
+      role.getEnvRunList().put("does.not.matter", new ArrayList<String>());
+   }
+
+   @Test(expectedExceptions = UnsupportedOperationException.class)
+   public void envRunListEntriesOnNewlyBuiltRoleIsImmutable() {
+      String env = "env";
+      Role role = Role.builder().envRunListElement(env, env + "1").build();
+      role.getEnvRunList().get(env).add("does.not.matter");
+   }
+
+   public void rolesWithSameEnvRunListAreEqual() {
+      String env = "env";
+      String entry = "entry";
+
+      Role role1 = Role.builder().envRunListElement(env, entry).build();
+      Role role2 = Role.builder().envRunListElement(env, entry).build();
+
+      assertEquals(role1.hashCode(), role2.hashCode(), "hashCodes should be equal");
+      assertEquals(role1, role2, "role1 should equal role2");
+      assertEquals(role2, role1, "role2 should equal role1");
+   }
+
+   public void rolesWithDifferentEnvRunListAreNotEqual() {
+      String env = "env";
+      String entry = "entry";
+
+      Role role1 = Role.builder().envRunListElement(env, entry.toUpperCase()).build();
+      Role role2 = Role.builder().envRunListElement(env, entry.toLowerCase()).build();
+
+      assertNotEquals(role1.hashCode(), role2.hashCode(), "hashCodes should not be equal");
+      assertNotEquals(role1, role2, "role1 should not equal role2");
+      assertNotEquals(role2, role1, "role2 should not equal role1");
+   }
+
+   private void verifyRunListForEnvironment(Map<String, List<String>> envRunList, String envName,
```

Comment:
Make this public static so you can call it from the live test too and avoid the code duplication?